UserContext/AnonymousRequestMatcher.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/*
+ * This file is part of the FOSHttpCacheBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCacheBundle\UserContext;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestMatcherInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Matches anonymous requests.
+ *
+ * Uses HTTP authentication headers as well as session name matching.
+ */
+class AnonymousRequestMatcher implements RequestMatcherInterface
+{
+    /**
+     * Valid options:
+     * - session_name_prefix.
+     *
+     * @var array
+     */
+    private $options;
+
+    public function __construct(array $options = [])
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults(['session_name_prefix' => 'PHPSESSID']);
+
+        $this->options = $resolver->resolve($options);
+    }
+
+    public function matches(Request $request)
+    {
+        if ($request->server->has('AUTHORIZATION') ||
+            $request->server->has('HTTP_AUTHORIZATION') ||
+            $request->server->has('PHP_AUTH_USER')
```

Comment:
i noticed that we have a configuration `user_identifier_headers`. maybe we should instead of looking for a session cookie or these headers just inject that list of headers and check for those headers.

the idea with cookies is that as we Vary anyways, the caching proxy has to clean up the cookie to not have random js cookie things in it. so we would not need to really go into the individual cookies.